### PR TITLE
Add new unset config API for Git implementation

### PR DESCRIPTION
**Pull request by offer8**
Add new unset config API for Git implementation

Change-Id: I601a4da51f70b0ab91510e0312ee0182f4e79c36

Signed-off-by: Offer Shostak offershostak@gmail.com

### CODE REVIEW

**Comment by offer8** [comment]
@tareqhs

**Comment by skabashnyuk** [comment]
Guys we really need some test.
Can you consider to add some everrest-assured mockito tests?
Like FactoryServiceTest

**Comment by offer8** [comment]
Hi,
We have a Git integration test file that checks all these APIs. We would like to push it after the JGitConnection pull request will be merged.
Is that OK?

Regards,
Offer.

**Comment by offer8** [comment]
Hi,

I've created new GitServiceTest class

Regards,
Offer.

**Comment by skabashnyuk** [comment]
Can I take a look on GitServiceTest?

**Comment by offer8** [comment]
Yes. It is in the pull request.

**Comment by skabashnyuk** [comment]
ok for me.
@vinokurig @tareqhs @sleshchenko ?

File: platform-api/che-core-api-git/src/test/java/org/eclipse/che/api/git/GitServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+import static com.jayway.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.containsString;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.when;
+import static org.testng.Assert.assertEquals;
+
+/**
+ * @author offer8
+ */
+@Listeners(value = { EverrestJetty.class, MockitoTestNGListener.class }) public class GitServiceTest {
```
Comment (sleshchenko) [inline platform-api/che-core-api-git/src/test/java/org/eclipse/che/api/git/GitServiceTest.java]: Strange formatting for me

```
@Listeners({EverrestJetty.class, MockitoTestNGListener.class })
public class GitServiceTest {
...
}
```

Looks much better

File: platform-api/che-core-api-git/src/test/java/org/eclipse/che/api/git/GitServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+    private final String SERVICE_PATH = "/git/ws1";
+    private final String PROJECT_PATH_PARAM_NAME = "projectPath";
+
+    @Mock private GitConnectionFactory gitConnectionFactory;
```
Comment (sleshchenko) [inline platform-api/che-core-api-git/src/test/java/org/eclipse/che/api/git/GitServiceTest.java]: Let's use common style for formatting in this case

```
@Mock 
private GitConnectionFactory gitConnectionFactory;
```

**Comment by offer8** [comment]
Fixed formatting.

**Comment by skabashnyuk** [comment]
Oftop. @offer8  why email of author of your commits and Signed-off-by is different?

File: platform-api/che-core-api-git/src/test/java/org/eclipse/che/api/git/GitServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,199 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2015 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ * <p/>
```
Comment (sleshchenko) [inline platform-api/che-core-api-git/src/test/java/org/eclipse/che/api/git/GitServiceTest.java]: broken license. Insert from GitService

**Comment by sleshchenko** [comment]
Other OK for me.
Note: It would be great commit these changes with email from your account on github

**Comment by offer8** [comment]
Fixed license and email comments.
The commit description contains my github email.

File: platform-api/che-core-api-git/src/main/java/org/eclipse/che/api/git/GitService.java [hunk 4]
```diff
@@ -426,6 +427,25 @@ public void setConfig(ConfigRequest request) throws ApiException {
         }
     }
 
+    @Path("config")
+    @DELETE
+    @Consumes(MediaType.APPLICATION_JSON)
```
Comment (vinokurig) [inline platform-api/che-core-api-git/src/main/java/org/eclipse/che/api/git/GitService.java]: This is not needed here and please remove it from `getConfig()` method

**Comment by offer8** [comment]
Removed.

**Comment by vinokurig** [comment]
OK, but license is still broken, please fix it

**Comment by offer8** [comment]
Now the license OK?

**Pull request merged**